### PR TITLE
MGTwoLevelTransfer: specialization for identity...

### PR DESCRIPTION
**Pull request by peterrum**
... and do the application of weights during reading/writing from/to the global vectors

### CODE REVIEW

File: include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h [hunk 1]
```diff
@@ -1649,6 +1649,36 @@ MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>::prolongate(
...
   for (const auto &scheme : schemes)
     {
+      // identity -> take short cut and work directly on global vectors
+      if (scheme.degree_fine == scheme.degree_coarse)
+        {
+          const unsigned int *indices_fine =
+            scheme.level_dof_indices_fine.data();
+          const unsigned int *indices_coarse =
+            scheme.level_dof_indices_coarse.data();
+          const Number *w = nullptr;
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h:1659]: Can we use slightly more verbose variable names?
```suggestion
          const Number *weights = nullptr;
```

File: include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h [hunk 2]
```diff
@@ -1692,29 +1722,26 @@ MGTwoLevelTransfer<dim, LinearAlgebra::distributed::Vector<Number>>::prolongate(
...
-                }
-            }
-
-
-          // write into dst vector
+          // weight and write into dst vector
           {
             const unsigned int *indices =
               &scheme.level_dof_indices_fine[cell * scheme.dofs_per_cell_fine];
+            const Number *w = nullptr;
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h:1729]:
```suggestion
            const Number *weights = nullptr;
```

**Review by kronbichler — APPROVED** [review]
Looks good apart from variable name.

File: include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h (hunk 1 above)
Comment (peterrum) [reply include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h]: Done!

**Review by kronbichler — APPROVED** [review]

**Pull request merged**